sample-app/src/hooks/useSynchronizedRequest.tsx

```diff
@@ -0,0 +1,24 @@
+import { useRef, useState } from "react";
+
+export function useSynchronizedRequest<RequestDataType, ResponseType>(
```

Comment:
Can you provide some JSDoc here? I'm not sure what this function does.

Reply:
Added!